include/zephyr/bluetooth/bluetooth.h

```diff
@@ -2049,6 +2049,9 @@ int bt_le_set_rpa_timeout(uint16_t new_rpa_timeout);
  * common scenario is to call this helper on the advertising data
  * received in the callback that was given to bt_le_scan_start().
  *
+ * @warning This helper function will consume `ad` when parsing. The user should
+ *          make a copy if the original data is to be used afterwards
```

Comment:
Maybe useful to add references to `net_buf_simple` APIs that can be used for this (clone or the save/restore).